legacy/jacorb/src/main/java/org/jboss/as/jacorb/JacORBSubsystemAdd.java

```diff
@@ -0,0 +1,166 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.jacorb;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Properties;
+
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.ProcessType;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.as.jacorb.logging.JacORBLogger;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.Property;
+import org.wildfly.iiop.openjdk.AttributeConstants;
+import org.wildfly.iiop.openjdk.Constants;
+import org.wildfly.iiop.openjdk.IIOPSubsystemAdd;
+
+import com.sun.corba.se.impl.orbutil.ORBConstants;
+
+/**
+ *
+ * @author <a href="mailto:tadamski@redhat.com">Tomasz Adamski</a>
+ */
+
+public class JacORBSubsystemAdd extends IIOPSubsystemAdd {
+
+    static final JacORBSubsystemAdd INSTANCE = new JacORBSubsystemAdd();
+
+    private JacORBSubsystemAdd() {
+        super(JacORBSubsystemDefinitions.SUBSYSTEM_ATTRIBUTES);
+    }
+
+    @Override
+    protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model)
+            throws OperationFailedException {
+        printJacORBEmulationWarningMessage();
+        super.performRuntime(context, operation, model);
+    }
+
+    private void printJacORBEmulationWarningMessage() {
+        JacORBLogger.ROOT_LOGGER.jacorbEmulationWarning();
+    }
+
+    @Override
+    protected void populateModel(final OperationContext context, final ModelNode operation, final Resource resource)
+            throws OperationFailedException {
+        super.populateModel(context, operation, resource);
+        final ModelNode model = resource.getModel();
+        if (!context.getProcessType().equals(ProcessType.HOST_CONTROLLER)) {
+            final List<String> propertiesToReject = new LinkedList<String>();
+            for (final AttributeDefinition attribute : JacORBSubsystemDefinitions.ON_OFF_ATTRIBUTES_TO_REJECT) {
+                if (model.hasDefined(attribute.getName())
+                        && model.get(attribute.getName()).equals(JacORBSubsystemDefinitions.DEFAULT_ENABLED_PROPERTY)) {
+                    propertiesToReject.add(attribute.getName());
+                }
+            }
+            for (final AttributeDefinition attribute : JacORBSubsystemDefinitions.ATTRIBUTES_TO_REJECT) {
+                if (model.hasDefined(attribute.getName())) {
+                    propertiesToReject.add(attribute.getName());
+                }
+            }
```

Comment:
Similar processing needs to happen with the write-attribute op; i.e. if any of these attributes get defined on a server via write-attribute, the op needs to fail.